clash-prelude/src/Clash/Signal/Time.hs

```diff
@@ -0,0 +1,40 @@
+{-|
+Copyright  :  (C) 2024 QBayLogic B.V.
+License    :  BSD2 (see the file LICENSE)
+Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
+
+User friendly utilities to express time in your digital design.
+
+-}
+module Clash.Signal.Time
```

Comment:
Do we really need a whole module for this? This makes the table of contents of the Haddock larger, so there's a trade-off. I feel they can go in `Clash.Signal.Internal`.

Reply:
Moved them :)